stream-chat-android-state/src/main/java/io/getstream/chat/android/offline/plugin/listener/internal/ThreadQueryListenerState.kt

```diff
@@ -59,35 +52,27 @@ internal class ThreadQueryListenerFull(
     }
 
     override suspend fun onGetRepliesRequest(messageId: String, limit: Int) {
-        val threadLogic = logic?.thread(messageId)
+        val threadLogic = logic.thread(messageId)
 
-        threadLogic?.setLoading(true)
-        val messages = messageRepository?.selectMessagesForThread(messageId, limit)
-        val parentMessage = threadLogic?.getMessage(messageId) ?: messages?.firstOrNull { it.id == messageId }
+        threadLogic.setLoading(true)
+        val messages = messageRepository.selectMessagesForThread(messageId, limit)
+        val parentMessage = threadLogic.getMessage(messageId) ?: messages.firstOrNull { it.id == messageId }
 
-        if (parentMessage != null && messages?.isNotEmpty() == true) {
-            threadLogic?.upsertMessages(messages)
-        } else {
-            val result = getRemoteMessage(messageId)
-            if (result.isSuccess) {
-                val message = result.data()
-                threadLogic?.upsertMessage(result.data())
-                userRepository?.insertUsers(message.users())
-                messageRepository?.insertMessage(message)
-            }
```

Comment:
Hey @leandroBorgesFerreira ,

I'm wondering where this `getRemoteMessage` block is moved to?

Reply:
I removed this part of the code. The remote message will be fetched from the backend once the result of the call is completed, then we don't need this call to be made in the pre call method. Also, making calls inside listeners is not good, because it may create a problem of a listener triggering a second listener which may create a loop. 

This changes the original behaviour of this listener, but I believe it is better to erase this call.